action.yml

```diff
@@ -71,14 +76,53 @@ runs:
         printenv CHALK_PUBLIC_KEY > chalk.pub
 
     - name: Save Private Key
-      if: inputs.private_key != ''
+      if: (runner.os == 'Linux' || runner.os == 'macOS') && inputs.private_key != ''
       shell: bash
       working-directory: ${{ github.action_path }}
       env:
         CHALK_PRIVATE_KEY: "${{ inputs.private_key }}"
       run: |
         printenv CHALK_PRIVATE_KEY > chalk.key
 
+    - name: Save provided JWT token
+      if: (runner.os == 'Linux' || runner.os == 'macOS') && inputs.token != ''
+      shell: bash
+      working-directory: ${{ github.action_path }}
+      run: |
+        echo "${{ inputs.token }}" > chalk.jwt
+
+    - name: Get JWT by using GitHub OpenId Connect
+      if: (runner.os == 'Linux' || runner.os == 'macOS') && inputs.connect != '' && inputs.connect != 'false' && inputs.token == ''
+      shell: bash
+      working-directory: ${{ github.action_path }}
+      run: |
+        curl \
+          --fail \
+          --silent \
+          --header "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" \
+          "$ACTIONS_ID_TOKEN_REQUEST_URL&audience=https://crashoverride.run" \
+          > github.jwt || (
+          echo "Cannot generate GitHub OpenId Connect JWT Token."
+          echo "Please make sure workflow/job has 'id-token: write' permission."
+          echo "See https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#adding-permissions-settings"
+          exit 1
+        )
+        curl \
+          --fail \
+          --silent \
+          --request POST \
+          --data-binary @github.jwt \
+          --header 'Content-Type: application/json' \
+          https://chalk.crashoverride.run/v0.1/openid-connect/github \
+          | jq -r '.jwt' \
+          > chalk.jwt || (
+          echo "Could not retrieve Chalk JWT token."
+          echo "Please make sure GitHub integration is configured in your workspace."
```

Comment:
tiny nitpick, possibly the error message could be extra specific and say `...in your Crash Override workspace.` so as there is no ambiguity of which workspace might be being referred to ?

Reply:
good idea. changing. thanks